datasync/kvdbsync/plugin_impl_dbsync.go

```diff
@@ -67,10 +77,12 @@ type Deps struct {
 }
 
 // Init only initializes plugin.registry.
-func (plugin *Plugin) Init() error {
-	plugin.registry = syncbase.NewRegistry()
+func (plugin *Plugin) Init() (err error) {
+	plugin.initOnce.Do(func() {
+		plugin.registry = syncbase.NewRegistry()
+	})
 
-	return nil
+	return err
```

Comment:
This is just confusing, `err` will always be `nil` here.

Reply:
It’s actually a protective pattern in this case.  The return value in this case comes from to us via closure on the func() being called by once.Do(). If someone subsequently alters the code so that there is an error, not having to remember to change this second return makes it less likely things will be screwed up.

Protect against future human error when it’s cheap.  It’s philosophically similar to putting defer close next to where you open something

Reply:
It’s actually a protective pattern in this case.  The return value in this case comes from to us via closure on the func() being called by once.Do(). If someone subsequently alters the code so that there is an error, not having to remember to change this second return makes it less likely things will be screwed up.

Protect against future human error when it’s cheap.  It’s philosophically similar to putting defer close next to where you open something

Reply:
Personally it would make me have to re-read the code to see where I missed `err` being updated. I think in this case returning `nil` would be preferred.

Reply:
I'd give a :+1: to `return nil` and a :-1: to `return err` if it's always going to be nil.